### PR TITLE
[SG-35750] Code monitoring create/edit page ARC checklist issues

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description
Resolves outlined accessibility defects

## Refs
[Sourcegraph Issue](https://github.com/sourcegraph/sourcegraph/issues/35750)
[GitStart Issue](https://app.gitstart.com/clients/sourcegraph/tickets/SG-35750)

## Implementation Details
- [x] The code monitoring logo should not have an `aria-label` on the a tag and have an `aria-label` on the `svg` tag reads "Code Monitoring"
- [x] The heading "Manage/create code monitor" should be h2 instead of h1 and "Trigger"/"Actions" headers should be h3 instead of h2. Also, the headings inside open cards (both trigger and actions) should be h4.
- [x] The owner should be labelled properly. Dropdown menu `select` tag for owner should have an ID and the `label` before it should have a `for` property to that ID. Also remove the aria-label for that select as it should no longer be needed.
- [x] Remove `aria-label` from closed trigger & action cards. Add a visually hidden text instead before the card's title (eg. "Send email notifications") that reads "Edit trigger:" or "Edit action:".
- [x] SVG icons in trigger query checklist should be aria-hidden
- [x] In the trigger card, icon next to "Preview results" should have `aria-label="Open in new window"`
- [x] In the Slack notifications card (NOT the webhooks card), rename label "Webhook URL" to "Slack webhook URL"
- [x] Remove title attribute from all toggles and ensure that `aria-labelledby` is pointing to a valid element (it is incorrect in some of them).

## Test plan
- A good way to test this PR is just to run the branch locally, and open [the issue link](https://github.com/sourcegraph/sourcegraph/issues/35750), compare the changes on the PR and the listed issues. All of them must be fixed

## App preview:

- [Web](https://sg-web-contractors-sg-35750.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-nhmmffxnbg.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by gitstart-sourcegraph** [comment]
Hi @limitedmage 
With regards to first requirement, we had initially hide the breadcrumb Icon([Ref here](https://github.com/sourcegraph/sourcegraph/pull/36627/files)), Should we override this default behavior only for this use case? 
Or Globally i.e Update the breadcrumb component to bind `ariaLabel` prop to `Icon` (only when `children` prop is not provided)?

**Comment by limitedmage** [comment]
Yes, I think this should be global (label should be on the icon itself, not the icon's wrapper) for all breadcrumbs that have icons. WDYT @sourcegraph/frontend-platform?

**Comment by umpox** [comment]
@limitedmage Agree. I also agree that we should only apply `ariaLabel` to `Icon` when `children` is not present

**Comment by gitstart-sourcegraph** [comment]
> @limitedmage Agree. I also agree that we should only apply `ariaLabel` to `Icon` when `children` is not present
> 
> @gitstart-sourcegraph Does that help?

That help @umpox 🙂, Thanks

File: client/web/src/enterprise/code-monitoring/components/actions/ActionEditor.tsx [hunk 1]
```diff
@@ -247,7 +246,6 @@ export const ActionEditor: React.FunctionComponent<React.PropsWithChildren<Actio
                             <div className="d-flex align-items-center">
                                 <div>
                                     <Toggle
-                                        title="Enabled"
                                         value={actionEnabled}
                                         onToggle={enabled => toggleActionEnabled(enabled, !expanded)}
```
Comment (gitstart-sourcegraph) [inline client/web/src/enterprise/code-monitoring/components/actions/ActionEditor.tsx:250]: Hi @limitedmage 
For use cases such as this, as part of task requirement we did removed the `title` property,
However there is no existing `aria-labelledby` (Unlike other use cases) property here, i.e screen reader has no alternative text to read for this element.
WDYT about an appropriate alternative text to be binded to this element for screen reader?

Comment (limitedmage) [reply client/web/src/enterprise/code-monitoring/components/actions/ActionEditor.tsx:250]: Ahh good catch. In this case, let's add `title={actionEnabled ? 'Enabled' : 'Disabled'}`

**Review by limitedmage — APPROVED** [review]

**Pull request merged**